value/listreflect.go

```diff
@@ -74,6 +74,19 @@ func (r listReflect) EqualsUsing(a Allocator, other List) bool {
 	return ListEqualsUsing(a, &r, other)
 }
 
+func toReflectValue(v Value) reflect.Value {
+	if vr, ok := v.(*valueReflect); ok {
+		return vr.Value
+	}
+	return reflect.ValueOf(v.Unstructured())
```

Comment:
I am using toUnstructured because the complementary function for maps does the same, but it makes me feel bad.

I feel like we should be using `FromUnstructured` but unfortunately that is in upstream k8s... @apelisse  tips?

Reply:
Wasn't referring to the performance cost. But setting any random reflect field to `Unstructured` sounds wrong...

Reply:
Yeah, I don't remember much but we discussed that with @jpbetz I think a long time ago, there's no way to make that work. If we receive an unstructured, we don't really know what types we're supposed to create, it's not made for that. We could get it through reflection but it's a lot more work. SMD only creates and modifies unstructured objects so that's easy but updating typed object wasn't part of the design.